src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverChatController.java

```diff
@@ -0,0 +1,150 @@
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.PageRequest;
+
+import javax.validation.Valid;
+
+import edu.ucsb.cs156.gauchoride.entities.DriverChat;
+import edu.ucsb.cs156.gauchoride.entities.User;
+import edu.ucsb.cs156.gauchoride.repositories.DriverChatRepository;
+import edu.ucsb.cs156.gauchoride.errors.EntityNotFoundException;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import java.time.LocalDateTime;
+
+@Api(description= "Driver chat information")
+@RequestMapping("/api/driverchats")
+@RestController
+public class DriverChatController extends ApiController {
+
+    @Autowired
+    DriverChatRepository driverChatRepository;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @ApiOperation(value = "Get a list of all chat message")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN','ROLE_DRIVER')")
+    @GetMapping("/all")
+    public Iterable<DriverChat> getAllMessages() {
+        return driverChatRepository.findAll();
+    }
+
+    @ApiOperation(value = "Get the latest N messages based on query parameters")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN','ROLE_DRIVER')")
+    @GetMapping("/list")
+    public Iterable<DriverChat> getRecentMessage(
+        @ApiParam(name = "limit", required = true, example ="100", value="an integer, representing how many messages to query", type="Long") @RequestParam int limit
+    ) {
+        Pageable pageable = PageRequest.of(0, limit);
+        return driverChatRepository.findAllByOrderByTimeStampDesc(pageable);
+    }
+
+    @ApiOperation(value = "Get a single chat message")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN','ROLE_DRIVER')")
+    @GetMapping("")
+    public DriverChat getMessageById(@ApiParam(name = "id", required = true, example="15", value="id of the message", type="Long") @RequestParam Long id) {
+        DriverChat message =  driverChatRepository.findById(id)
+        .orElseThrow(()->new EntityNotFoundException(DriverChat.class, id));
+
+        return message;
+    }
+
+    @ApiOperation(value="Create a new chat message")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN','ROLE_DRIVER')")
+    @PostMapping("/post")
+    public DriverChat postNewMessage(
+        @RequestBody String messageContent
+    )  throws JsonProcessingException {
+
+        User currentUser = getCurrentUser().getUser();
+        DriverChat chatMessage = DriverChat.builder()
+        .sender(currentUser)
+        .messageContent(messageContent)
+        .timeStamp(LocalDateTime.now())
+        .build();
+      
+        return  driverChatRepository.save(chatMessage);
+    }
+
+    @ApiOperation(value="Update a driver's message")
+    @PreAuthorize("hasRole('ROLE_DRIVER')")
```

Comment:
I think this will make more sense in the long run.   An admin should also be able to delete their own messages; they shouldn't need to go to a different page to use their "admin powers" to delete their own messages.  What do you think?

```suggestion
    @PreAuthorize("hasAnyRole('ROLE_DRIVER', 'ROLE_ADMIN')")
```